omigami/feature_selector.py

```diff
@@ -22,68 +37,198 @@ def __init__(
+                    input_data,
+                    outer_split,
+                    executor=executor,
+                    data_splitter=data_splitter,
+                )
+                outer_loop_results.append(outer_loop_result)
+            repetition_results.append(outer_loop_results)
+
+        self._selected_features = self._select_best_features(repetition_results)
         self.is_fit = True
         return self
 
-    def _execute_repetitions(self, outer_loop: OuterLoop) -> List[Repetition]:
-        results = []
-        for _ in range(self.repetitions):
-            outer_loop.refresh_splits()
-            result = outer_loop.run(executor=self.executor)
-            results.append(result)
-        return results
+    @staticmethod
+    def get_groups(groups: NumpyArray, size: int) -> NumpyArray:
+        if groups is None:
+            logging.info("groups is not specified: i.i.d. samples assumed")
+            groups = np.arange(size)
+        return groups
 
-    def _make_feature_evaluator(self, input_data: InputData) -> FeatureEvaluator:
-        return FeatureEvaluator(
-            input_data,
-            self.n_outer,
-            self.n_inner,
-            self.estimator,
-            self.metric,
-            self.random_state,
+    def _deferred_run_outer_loop(
+        self,
+        input_data: InputDataset,
+        outer_split: Split,
+        data_splitter: DataSplitter,
+        executor: Executor,
+    ) -> Union[Future, OuterLoopResults]:
+        if executor is None:
+            return self._run_outer_loop(input_data, outer_split, data_splitter)
+        return executor.submit(
+            self._run_outer_loop, input_data, outer_split, data_splitter
+        )
+
+    def _run_outer_loop(
+        self, input_data: InputDataset, outer_split: Split, data_splitter: DataSplitter,
+    ) -> OuterLoopResults:
+
+        raw_feature_elim_results = {}
+        feature_set = list(range(input_data.n_features))
+
+        while len(feature_set) >= self._minimum_features:
+            inner_results = []
+
+            for inner_split in data_splitter.iter_inner_splits(outer_split):
+                inner_loop_data = data_splitter.split_data(
+                    input_data, inner_split, feature_set
+                )
+
+                feature_evaluation_results = self.feature_evaluator.evaluate_features(
+                    inner_loop_data, feature_set
+                )
+
+                inner_results.append(feature_evaluation_results)
+
+            raw_feature_elim_results[tuple(feature_set)] = inner_results
+            feature_set = self._remove_features(feature_set, inner_results)
+
+        feature_elimination_results = self.post_processor.process_feature_elim_results(
+            raw_feature_elim_results
+        )
+        outer_loop_results = self.create_outer_loop_results(
+            feature_elimination_results, input_data, outer_split, data_splitter
```

Comment:
`feature_elimination_results` is just passed through, adding complexity to this interface can this be avoided?

Reply:
:+1: